thoth/storages/graph/postgres.py

```diff
@@ -1880,6 +1880,78 @@ def get_analyzed_error_python_package_versions_count_all(
 
     # Unanalyzed Python Packages
 
+    def _construct_si_unanalyzed_python_package_versions_query(
+        self, session: Session, os_name: str = None, os_version: str = None, python_version: str = None, index_url: str = None
+    ) -> Query:
+        """Construct query for packages analyzed by solver, but unanalyzed by SI."""
+        index_url = GraphDatabase.normalize_python_index_url(index_url)
+        query = session.query(PythonPackageVersion).filter(
+            PythonPackageVersion.package_version.isnot(None),
+            PythonPackageIndex.url.isnot(None),
+            PythonPackageIndex.enabled.is_(True),
+        )
+
+        if index_url is not None:
+            query = query.filter(PythonPackageIndex.url == index_url)
+
+        # We find all rows that are same in PythonPackageVersion and SIAggregated table.
+        conditions = [PythonPackageVersion.entity_id == SIAggregated.id]
+
+        if os_name:
+            conditions.append(PythonPackageVersion.os_name == os_name)
+
+        if os_version:
+            conditions.append(PythonPackageVersion.os_version == os_version)
+
+        if python_version:
+            conditions.append(
+                PythonPackageVersion.python_version == python_version)
+
+        # Finally filter these out.
+        query = query.filter(~exists().where(and_(*conditions)))
+
+        return query
+
+    def get_si_unanalyzed_python_package_versions_all(
+        self,
+        os_name: str = None,
+        os_version: str = None,
+        python_version: str = None,
+        start_offset: int = 0,
+        count: Optional[int] = DEFAULT_COUNT,
+        distinct: bool = False,
```

Comment:
Do we want to have distinct by default?

```suggestion
        distinct: bool = True,
```

Reply:
Do we want the same behavior in `get_unsolved_python_packages_version_all` ?